src/main/java/com/fabriccommunity/spookytime/world/feature/LargeSkeletalTreeFeature.java

```diff
@@ -0,0 +1,91 @@
+package com.fabriccommunity.spookytime.world.feature;
+
+import com.fabriccommunity.spookytime.registry.SpookyBlocks;
+import com.mojang.datafixers.Dynamic;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.PillarBlock;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.util.math.MutableIntBoundingBox;
+import net.minecraft.world.ModifiableTestableWorld;
+import net.minecraft.world.TestableWorld;
+import net.minecraft.world.gen.feature.AbstractTreeFeature;
+import net.minecraft.world.gen.feature.DefaultFeatureConfig;
+
+import java.util.Iterator;
+import java.util.Random;
+import java.util.Set;
+import java.util.function.Function;
+
+/**
+ * @author Indigo Amann
+ */
+public class LargeSkeletalTreeFeature extends AbstractTreeFeature<DefaultFeatureConfig> {
+    private static final BlockState LOG;
+
+    public LargeSkeletalTreeFeature(Function<Dynamic<?>, ? extends DefaultFeatureConfig> function, boolean boolean1) {
+        super(function, boolean1);
+    }
+
+    private void addLog(Set<BlockPos> set, ModifiableTestableWorld modifiableTestableWorld, BlockPos blockPos, Direction.Axis axis, MutableIntBoundingBox mutableIntBoundingBox) {
+        if (canTreeReplace(modifiableTestableWorld, blockPos)) {
+            this.setBlockState(set, modifiableTestableWorld, blockPos, LOG.with(PillarBlock.AXIS, axis), mutableIntBoundingBox);
+        }
+
+    }
+
+    private static boolean isPlantableOn(TestableWorld testableWorld, BlockPos blockPos) {
+        return testableWorld.testBlockState(blockPos, (blockState) -> {
+            Block block = blockState.getBlock();
+            return Block.isNaturalDirt(block) || block == Blocks.GRASS_BLOCK || block == SpookyBlocks.DECEASED_GRASS_BLOCK || block == Blocks.BONE_BLOCK;
+        });
+    }
+
+    private static boolean isSurroundedByAir(TestableWorld testableWorld, BlockPos blockPos, Direction ignore) {
+        for (Iterator<Direction> it = Direction.Type.HORIZONTAL.iterator(); it.hasNext(); ) {
+            Direction direction = it.next();
+            if (direction == ignore) continue;
+            if (!isAir(testableWorld, blockPos.offset(direction))) return false;
+        }
+        return true;
+    }
+
+    static {
+        LOG = Blocks.BONE_BLOCK.getDefaultState();
+    }
```

Comment:
I would say we can just move this up to where the field is declared-- no reason for a separate static block.

Reply:
I would also agree here.